arch/arm/core/aarch64/reset.S

```diff
@@ -119,39 +118,10 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	adr	x0, 1f
 	msr	elr_el3, x0
 	eret
-#endif
-	/*
-	 * Enable the instruction cache, stack pointer and data access
-	 * alignment checks.
-	 */
-	mov	x1, #(SCTLR_I_BIT | SCTLR_A_BIT | SCTLR_SA_BIT)
-	mrs	x0, sctlr_el3
-	orr	x0, x0, x1
-	msr	sctlr_el3, x0
-	isb
-	b	0f
 
 2:
-	/* Initialize VBAR */
-	msr	vbar_el2, x19
-
-	/* SError, IRQ and FIQ routing enablement in EL2 */
-	mrs	x0, hcr_el2
-	orr	x0, x0, #(HCR_EL2_FMO | HCR_EL2_IMO | HCR_EL2_AMO)
-	msr	hcr_el2, x0
-
-	/* Disable access trapping in EL2 for NEON/FP */
-	msr	cptr_el2, xzr
-
-	/*
-	 * Enable the instruction cache, stack pointer and data access
-	 * alignment checks.
-	 */
-	mov	x1, #(SCTLR_I_BIT | SCTLR_A_BIT | SCTLR_SA_BIT)
-	mrs	x0, sctlr_el2
-	orr	x0, x0, x1
-	msr	sctlr_el2, x0
-	b	0f
+	/* Booting from EL2 is not supported */
+	b	.
```

Comment:
I am bit skeptical about this (b .) as user will not have any idea where system hung. But I do not have alternate idea at this point. (there may be some idea around the fact that we should be able to access EL1 registers from EL2 if accessing and assert at later point where we are able to print the fault)

Everything else looks fine.